Mathlib/CategoryTheory/MorphismProperty/Composition.lean

```diff
@@ -89,39 +95,16 @@ theorem StableUnderInverse.unop {P : MorphismProperty Cᵒᵖ} (h : StableUnderI
     StableUnderInverse P.unop := fun _ _ e he => h e.op he
 #align category_theory.morphism_property.stable_under_inverse.unop CategoryTheory.MorphismProperty.StableUnderInverse.unop
 
-theorem StableUnderComposition.respectsIso {P : MorphismProperty C} (hP : StableUnderComposition P)
-    (hP' : ∀ {X Y} (e : X ≅ Y), P e.hom) : RespectsIso P :=
-  ⟨fun e _ hf => hP _ _ (hP' e) hf, fun e _ hf => hP _ _ hf (hP' e)⟩
-#align category_theory.morphism_property.stable_under_composition.respects_iso CategoryTheory.MorphismProperty.StableUnderComposition.respectsIso
-
-theorem StableUnderComposition.isomorphisms : StableUnderComposition (isomorphisms C) :=
-  fun X Y Z f g hf hg => by
-  rw [isomorphisms.iff] at hf hg ⊢
-  haveI := hf
-  haveI := hg
-  infer_instance
-#align category_theory.morphism_property.stable_under_composition.isomorphisms CategoryTheory.MorphismProperty.StableUnderComposition.isomorphisms
-
-theorem StableUnderComposition.monomorphisms : StableUnderComposition (monomorphisms C) :=
-  fun X Y Z f g hf hg => by
-  rw [monomorphisms.iff] at hf hg ⊢
-  haveI := hf
-  haveI := hg
-  apply mono_comp
-#align category_theory.morphism_property.stable_under_composition.monomorphisms CategoryTheory.MorphismProperty.StableUnderComposition.monomorphisms
-
-theorem StableUnderComposition.epimorphisms : StableUnderComposition (epimorphisms C) :=
-  fun X Y Z f g hf hg => by
-  rw [epimorphisms.iff] at hf hg ⊢
-  haveI := hf
-  haveI := hg
-  apply epi_comp
-#align category_theory.morphism_property.stable_under_composition.epimorphisms CategoryTheory.MorphismProperty.StableUnderComposition.epimorphisms
-
-theorem StableUnderComposition.inverseImage {P : MorphismProperty D} (h : StableUnderComposition P)
-    (F : C ⥤ D) : StableUnderComposition (P.inverseImage F) := fun X Y Z f g hf hg => by
-  simpa only [← F.map_comp] using h (F.map f) (F.map g) hf hg
-#align category_theory.morphism_property.stable_under_composition.inverse_image CategoryTheory.MorphismProperty.StableUnderComposition.inverseImage
+theorem respectsIso_of_isStableUnderComposition {P : MorphismProperty C}
+    [P.IsStableUnderComposition] (hP : isomorphisms C ⊆ P) :
```

Comment:
I noticed that a lot of these are formalized with subset rather than le. Is there a reason to this? It will require either a lot more duplication or rewrites into le when using the order API (when considering the disjunction of properties, or monotone property constructors etc.)
(out of scope of this PR)

Reply:
For `MorphismProperty`, I prefer `⊆` over `≤`, because if `h : W₁ ⊆ W₂`, it can be used as `h f hf`, rather than `h X Y f hf` if it were `h : W₁ ≤ W₂` (and `f : X ⟶ Y`).

Reply:
Hmmm good point. I'll try later if I can change up the implicitness when deriving the complete lattice instance.